crates/publish/src/oci/client.rs

```diff
@@ -248,3 +314,94 @@ impl Client {
         }
     }
 }
+
+#[derive(Serialize, Deserialize)]
+struct AuthConfig {
+    /// Map between registry server and base64 encoded username:password credential set.
+    pub auths: HashMap<String, String>,
+}
+
+impl AuthConfig {
+    /// Load the authentication configuration from the default location
+    /// ($XDG_CONFIG_HOME/fermyon/registry-auth.json).
+    pub async fn load_default() -> Result<Self> {
+        // TODO: add a way to override this path.
+        let default_path = dirs::config_dir()
+            .context("Cannot find configuration directory")?
+            .join("fermyon")
```

Comment:
Shouldn't this be "spin"?

Reply:
This is where the Cloud auth file is.. really don't want to split to yet another directory before we consolidate..

Reply:
Well, I think that's where it ultimately belongs and we'll have to do some kind of migration if we want to move it later...

Reply:
When we moved things to the `fermyon`, we talked about potentially having multiple Fermyon tools sharing that configuration.

I still think that is a good idea, so moving them under `spin` wouldn't really make sense. That being said, without a proposal for consolidating configuration..